### DISCUSSION TITLE
Training with pretrained weights file requires raw text to be present?

### DISCUSSION THREAD

**Discussion by erotavlas** [explosion/spaCy#8123] [Help: Coding & Implementations] [answered]
I was trying to train my ner model using the pretrained weights and the pretraining configuration file.  Hoever the file on it's own didn't work I kept getting the following error

```
=========================== Initializing pipeline ===========================
✘ Config validation error
corpora.pretrain -> path   none is not an allowed value
```

I specified the raw text in my command line as opposed to the config file like this, so I guess it wasn't added to the config?

`python -m spacy pretrain config_pretrain.cfg output_pretrain --paths.raw_text rawtext_preprocessed.jsonl -r "output_pretrain\model219.bin" --epoch-resume 219 --gpu-id 0 --code tokenizer_rules.py`

Then when I tried to use that config for training it gave the above error.  So to get around it I had to copy the raw text file onto the machine I was training and specify the same parameter for the raw text 

` --paths.raw_text rawtext_preprocessed.jsonl `

**Comment by polm** [comment]
So to clarify, you are using pretraining, but if you use the pretraining config to do normal training, it complains that you don't have the pretraining data (even though it shouldn't be using it). 

I was able to reproduce this issue. It seems weird to me, but I think it may be a side effect of the type checking in the config. 

An easy workaround, indirectly recommended by [the pretraining docs](https://spacy.io/usage/embeddings-transformers#pretraining), is to have a pretraining-specific config file, and use the base config for normal training.

I'll check with the team on whether this is intentional, and if we can change it so this isn't an issue.

**Comment by svlandeg** [comment, marked as answer]
We could have a look into fixing this to prevent the `[pretraining]` and `[corpora.pretrain]` blocks being resolved at training time.

> So to get around it I had to copy the raw text file onto the machine I was training and specify the same parameter for the raw text

The good news is that there's an easy workaround. The type checking complains if the `raw_text` is `None`, but this value isn't actually used / parsed during training. That means you won't have to actually copy the raw text file over. You can just keep the same command as you had `--paths.raw_text rawtext_preprocessed.jsonl` or even something silly like `--paths.raw_text="randomstring"` when training, and it'll pass, because it won't actually need that file at that stage.

I wouldn't create two separate configs for training & pretraining though. The idea is definitely to use the same config file for both steps, to ensure all settings stay in sync.
